cider/exec/template/CompareIR.cpp

```diff
@@ -339,14 +339,17 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenCmpFun(
 
   if (lhs_nullable && rhs_nullable) {
     if (lhs_nullable->getNull() && rhs_nullable->getNull()) {
-      null = cgen_state_->ir_builder_.CreateAnd(lhs_nullable->getNull(),
-                                                rhs_nullable->getNull());
+      null = cgen_state_->ir_builder_.CreateOr(lhs_nullable->getNull(),
+                                               rhs_nullable->getNull());
     } else {
       null = lhs_nullable->getNull() ? lhs_nullable->getNull() : rhs_nullable->getNull();
     }
   } else if (lhs_nullable || rhs_nullable) {
     null = lhs_nullable ? lhs_nullable->getNull() : rhs_nullable->getNull();
   }
+  if (lhs_ti.is_string()) {
```

Comment:
will it always be `lhs_ti` or it also can be `rhs_ti`?

Reply:
I think for a compare BinOper, left and right child would be same type.

Reply:
Suggest to use ```switch``` to dispatch methods of different types.